### PR TITLE
MT46629 : hoursPerDay display

### PR DESCRIPTION
**Pull request by lau-fst**
- deleted display on comptime (both day mode and hour mode)
 - bugfix : conversion now only on hour mode, no more display contradiction
 - simplified condition in template
 - added `ajax.holidays-hours-per-day` route to return hoursPerDay values for a specific agent
 - added the `resetHoursPerDay()` fct to update the hoursPerDay value when the selected agent changes

### CODE REVIEW

File: src/Controller/HolidayController.php [hunk 1]
```diff
@@ -274,6 +274,20 @@ public function index(Request $request, Session $session): Response
         return $this->output('holiday/index.html.twig');
     }
 
+    #[Route(path: '/ajax/holidays-hours-per-day', name: 'ajax.holidays-hours-per-day', methods: ['GET'])]
```
Comment (jeromecombes) [inline src/Controller/HolidayController.php]: Renome la route de cette façon : 
* path : /holiday/hours-per-day
* name holiday.hours-per-day

Modifie les fichiers permissions.yaml et holiday.js en conséquence.

File: src/Controller/HolidayController.php [hunk 2]
```diff
@@ -274,6 +274,20 @@ public function index(Request $request, Session $session): Response
...
+    public function blip(Request $request, Session $session) : \Symfony\Component\HttpFoundation\JsonResponse
+    {
+        $perso_id = $request->get('id');
```
Comment (jeromecombes) [inline src/Controller/HolidayController.php]: utilise `$perso_id = $request->query->getInt('id');`

File: src/Controller/HolidayController.php [hunk 3]
```diff
@@ -274,6 +274,20 @@ public function index(Request $request, Session $session): Response
         return $this->output('holiday/index.html.twig');
     }
 
+    #[Route(path: '/ajax/holidays-hours-per-day', name: 'ajax.holidays-hours-per-day', methods: ['GET'])]
+    public function blip(Request $request, Session $session) : \Symfony\Component\HttpFoundation\JsonResponse
```
Comment (jeromecombes) [inline src/Controller/HolidayController.php]: Renome la fonction en getHoursPerDay
Supprime Session $session (si la session n'est pas utilisée dans la fonction)
Ajoute use Symfony\Component\HttpFoundation\JsonResponse; au début du fichier, tu pourras ici simplifier en mettant seulement JsonResponse

`public function getHoursPerDay(Request $request) : JsonResponse`

File: src/Planno/Helper/HolidayHelper.php [hunk 4]
```diff
@@ -262,7 +262,7 @@ public function getCountedHours(): array
 
     public function hoursPerDay($perso_id, $holidays_hours_per_year = null)
     {
-        if ($this->config('conges-hours-per-day')) {
+        if ($this->config['Conges-Mode'] == 'heures' and $this->config('conges-hours-per-day')) {
```
Comment (jeromecombes) [inline src/Planno/Helper/HolidayHelper.php]: `if ($this->config['Conges-Mode'] == 'heures' and $this->config['conges-hours-per-day']) {`

File: src/Planno/Helper/HolidayHelper.php [hunk 5]
```diff
@@ -276,17 +276,24 @@ public function hoursPerDay($perso_id, $holidays_hours_per_year = null)
...
 
     public function hoursToDays($given_hours, $perso_id, $holidays_hours_per_year = null, $human_readable = false, $prefix = null) {
         if (empty($given_hours) and !$human_readable) { return 0; }
         if (empty($given_hours) and $human_readable) { return ''; }
 
-        $hours_per_day = ($holidays_hours_per_year == null) ? $this->hoursPerDay($perso_id) : $this->hoursPerDay(null, $holidays_hours_per_year);
+        $hours_per_day = 7;
 
-        $result = round((float) $given_hours / $hours_per_day, 2);
+        if ($this->config('Conges-Mode') == 'heures') {
```
Comment (jeromecombes) [inline src/Planno/Helper/HolidayHelper.php]: `if ($this->config['Conges-Mode'] == 'heures') {`

File: public/js/holiday.js [hunk 6]
```diff
@@ -188,6 +188,28 @@ function resetTerms(){
...
+
+    $.ajax({
+      url: url('ajax/holidays-hours-per-day'),
+      data: { id: perso_id },
+      dataType: 'json',
+      type: 'get',
+      async: false,
+      success: function(result){
+        $('#hours_per_day').val(result['hoursPerDay']);
+        $('label[for=nbJours]').html('Équivalence en jour :<br/>(1 jour = '+ result['hoursPerDayInHoursMinutes'] +')')
```
Comment (jeromecombes) [inline public/js/holiday.js]: Il serait bien de ne modifier que la valeur.

Essaye ceci
dans le template
`<label class="col-2 col-form-label text-end" for="nbJours">Équivalence en jour :<br/>(1 jour =  <font id="hours-per-day">{{ hours_per_day_in_hhmm }}</font>)</label>`

Ici 
`$('#hours-per-day').text(result['hoursPerDayInHoursMinutes']);`

N'oublie pas le ;

**Review by jeromecombes — CHANGES_REQUESTED** [review]

**Review by jeromecombes — APPROVED** [review]

**Comment by jeromecombes** [comment]
Merged on master (7786e09cdbcb9cbce0e95ae8cc8dbfc64434df9f)
